### PR TITLE
refactor/203: updates base repository to support combining specificat…

### PR DESCRIPTION
**Pull request by IrynaNakoneshniuk**
…ions

dev
## JIRA

* [Issue 203](https://github.com/project-studying-dotnet/Streetcode-Server-June/issues/203)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

- Updated the base repository to support combining specifications.

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: Streetcode/Streetcode.DAL/Repositories/Interfaces/Base/IRepositoryBase.cs [hunk 1]
```diff
@@ -38,17 +38,17 @@ public interface IRepositoryBase<T>
     Task<IEnumerable<T>> GetAllAsync(
         Expression<Func<T, bool>>? predicate = default,
         Func<IQueryable<T>, IIncludableQueryable<T, object>>? include = default);
-    Task<IEnumerable<T>?> GetAllAsync(ISpecification<T> specification);
 
     Task<IEnumerable<T>?> GetAllAsync(
         Expression<Func<T, T>> selector,
         Expression<Func<T, bool>>? predicate = default,
         Func<IQueryable<T>, IIncludableQueryable<T, object>>? include = default);
+    Task<IEnumerable<T>?> GetAllWithSpecAsync(params ISpecification<T>[] specs);
```
Comment (Careless-boop) [inline Streetcode/Streetcode.DAL/Repositories/Interfaces/Base/IRepositoryBase.cs:46]: Why you added WithSpec in naming? Methods with different parameters can have the same naming

**Review by Careless-boop — APPROVED** [review]

File: Streetcode/Streetcode.DAL/Repositories/Interfaces/Base/IRepositoryBase.cs [hunk 2]
```diff
@@ -38,17 +38,17 @@ public interface IRepositoryBase<T>
...
 
     Task<T?> GetSingleOrDefaultAsync(
         Expression<Func<T, bool>>? predicate = default,
         Func<IQueryable<T>, IIncludableQueryable<T, object>>? include = default);
-    Task<T?> GetSingleOrDefaultAsync(ISpecification<T> specification);
+    Task<T?> GetSingleOrDefaultWithSpecAsync(params ISpecification<T>[] specs);
```
Comment (Careless-boop) [inline Streetcode/Streetcode.DAL/Repositories/Interfaces/Base/IRepositoryBase.cs:51]: Same here

File: Streetcode/Streetcode.DAL/Repositories/Interfaces/Base/IRepositoryBase.cs (hunk 1 above)
Comment (IrynaNakoneshniuk) [reply Streetcode/Streetcode.DAL/Repositories/Interfaces/Base/IRepositoryBase.cs:46]: The above method that has parameters with default values in its signature. If you use both params and default parameters , the compiler will not be able to determine which method to use

File: Streetcode/Streetcode.DAL/Repositories/Interfaces/Base/IRepositoryBase.cs (hunk 2 above)
Comment (IrynaNakoneshniuk) [reply Streetcode/Streetcode.DAL/Repositories/Interfaces/Base/IRepositoryBase.cs:51]: Answer above

**Review by vladtymo — APPROVED** [review]

**Pull request merged**